{"has_code": true, "source_code": "package original;\nimport java.util.ArrayList;\nimport org.junit.Test;\nimport static org.junit.Assert.*;\nimport java.util.List;\nimport java.util.Arrays;\n\npublic class TriTest {\n\n    @Test\n    public void testTriWithZero() {\n        List<Number> expected = Arrays.asList(1);\n        List<Number> result = Tri.tri(0);\n        assertEquals(expected, result);\n    }\n\n    @Test\n    public void testTriWithPositiveNumber() {\n        List<Number> expected = Arrays.asList(1, 3, 4.5, 7.0, 10.5);\n        List<Number> result = Tri.tri(4);\n        assertEquals(expected, result);\n    }\n\n    @Test\n    public void testTriWithNegativeNumber() {\n        List<Number> expected = Arrays.asList(1, 3, 4.5, 7.0, 10.5, 15.0, 20.5);\n        List<Number> result = Tri.tri(-6);\n        assertEquals(expected, result);\n    }\n}", "has_syntactic_error": false, "runtime_error": "\nFailures (2):\n  JUnit Vintage:Tri_20_1Test:testTriWithNegativeNumber\n    MethodSource [className = 'original.Tri_20_1Test', methodName = 'testTriWithNegativeNumber', methodParameterTypes = '']\n    => java.lang.AssertionError: expected:<[1, 3, 4.5, 7.0, 10.5, 15.0, 20.5]> but was:<[1, 3]>\n       org.junit.Assert.fail(Assert.java:89)\n       org.junit.Assert.failNotEquals(Assert.java:835)\n       org.junit.Assert.assertEquals(Assert.java:120)\n       org.junit.Assert.assertEquals(Assert.java:146)\n       original.Tri_20_1Test.testTriWithNegativeNumber(Tri_20_1Test.java:28)\n       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n       java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n       java.base/java.lang.reflect.Method.invoke(Method.java:566)\n       org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)\n       [...]\n  JUnit Vintage:Tri_20_1Test:testTriWithPositiveNumber\n    MethodSource [className = 'original.Tri_20_1Test', methodName = 'testTriWithPositiveNumber', methodParameterTypes = '']\n    => java.lang.AssertionError: expected:<[1, 3, 4.5, 7.0, 10.5]> but was:<[1, 3, 2.0, 8.0, 3.0]>\n       org.junit.Assert.fail(Assert.java:89)\n       org.junit.Assert.failNotEquals(Assert.java:835)\n       org.junit.Assert.assertEquals(Assert.java:120)\n       org.junit.Assert.assertEquals(Assert.java:146)\n       original.Tri_20_1Test.testTriWithPositiveNumber(Tri_20_1Test.java:21)\n       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n       java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n       java.base/java.lang.reflect.Method.invoke(Method.java:566)\n       org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)\n       [...]\n\nTest run finished after 44 ms\n[         4 containers found      ]\n[         0 containers skipped    ]\n[         4 containers started    ]\n[         0 containers aborted    ]\n[         4 containers successful ]\n[         0 containers failed     ]\n[         3 tests found           ]\n[         0 tests skipped         ]\n[         3 tests started         ]\n[         0 tests aborted         ]\n[         1 tests successful      ]\n[         2 tests failed          ]\n\n"}